src/core.js

```diff
@@ -425,6 +425,16 @@ jQuery.extend({
 	support: support
 });
 
+// jsHint would error on this code due to the Symbol not being defined in ES5.
+// Defining this global in .jshintrc would create a danger of using the global
+// unguarded in another place, it seems safer to just disable jsHint for these
+// three lines.
+/* jshint ignore: start */
+if ( typeof Symbol === "function" && Symbol.iterator ) {
```

Comment:
> Many people still support IE 8, so people will be using Symbol polyfills for a very long time, including with jQuery v3.0.0.

I realise that; however, people test on modern Chrome/Firefox more than they test on IE8 so I don't think people will write code expecting a polyfill and not discovering it breaks in browsers with native `Symbol`, rather the opposite.

Bad polyfills can do a lot of damage even if we try to protect against them so I'd try to go this route and see if we get any feedback about possible problems with this approach rather than lock down the code to strictly native symbols - in that latter case people would rely on the iterability only in Chrome extensions, Firefox OS apps, Windows 10 apps, Node/io.js etc. but not on the Web. Or even worse - they may use a polyfill but test only in Chrome/Firefox not knowing that the polyfill is explicitly excluded by our strict logic.

So I'd follow @dmethvin's advice and just see where the proposed approach leads us. :)